web/src/engine/common/web-utils/src/tests/stringHandling.js

```diff
@@ -1,342 +1,342 @@
-      // assert.equal(str._kmwSlice(-4), "dog.");
-      assert.equal(str._kmwSlice(-9, -5), "lazy");
+      // assert.equal(KMWString.slice(str, -4), "dog.");
+      assert.equal(KMWString.slice(str, -9, -5), "lazy");
       // FAILS:  does not handle `undefined` start like the real .slice() method.
-      // assert.equal(str._kmwSlice(), str);
+      // assert.equal(KMWString.slice(str), str);
 
       // Per documented spec for slice...
-      assert.equal(str._kmwSlice(31, -5), "the lazy"); // " dog." = 5 chars
-      assert.equal(str._kmwSlice(-5, -9), "");
+      assert.equal(KMWString.slice(str, 31, -5), "the lazy"); // " dog." = 5 chars
+      assert.equal(KMWString.slice(str, -5, -9), "");
     });
 
-    it('_kmwSubstr', () => {
+    it('substr', () => {
       const str = "The quick brown fox jumps over the lazy dog.";
 
-      assert.equal(str._kmwSubstr(1000), '');
-      assert.equal(str._kmwSubstr(31), "the lazy dog.");
-      assert.equal(str._kmwSubstr(-4), "dog.");
-      assert.equal(str._kmwSubstr(31, 31), "the lazy dog.");
-      assert.equal(str._kmwSubstr(31, -5), ''); // The big difference from .slice:  length must be non-negative.
+      assert.equal(KMWString.substr(str, 1000), '');
+      assert.equal(KMWString.substr(str, 31), "the lazy dog.");
+      assert.equal(KMWString.substr(str, -4), "dog.");
+      assert.equal(KMWString.substr(str, 31, 31), "the lazy dog.");
+      assert.equal(KMWString.substr(str, 31, -5), ''); // The big difference from .slice:  length must be non-negative.
     });
 
-    it('_kmwSubstring', () => {
+    it('substring', () => {
       const str = "The quick brown fox jumps over the lazy dog.";
 
       // FAILS - doesn't coerce the out-of-bounds value.
-      // assert.equal(str._kmwSubstring(1000), '');
-      assert.equal(str._kmwSubstring(31), "the lazy dog.");
-      assert.equal(str._kmwSubstring(31, 1000), "the lazy dog.");
-      assert.equal(str._kmwSubstring(4, 9), "quick");
+      // assert.equal(KMWString.substring(str, 1000), '');
+      assert.equal(KMWString.substring(str, 31), "the lazy dog.");
+      assert.equal(KMWString.substring(str, 31, 1000), "the lazy dog.");
+      assert.equal(KMWString.substring(str, 4, 9), "quick");
 
       // negatives are coerced to 0 & parameters are used in order of ascending value.
-      assert.equal(str._kmwSubstring(9, 4), "quick");
+      assert.equal(KMWString.substring(str, 9, 4), "quick");
       // FAILS:  these do not handle negative-valued inputs the same way the real .substring does.
-      // assert.equal(str._kmwSubstring(31, -4), str.substring(0, 31));
-      // assert.equal(str._kmwSubstring(-5, 9), str.substring(0, 9));
-      // assert.equal(str._kmwSubstring(-9, -5), '');
+      // assert.equal(KMWString.substring(str, 31, -4), str.substring(0, 31));
+      // assert.equal(KMWString.substring(str, -5, 9), str.substring(0, 9));
+      // assert.equal(KMWString.substring(str, -9, -5), '');
```

Comment:
Looks to me we should enable these tests and make `KMWString.substring` compatible with `string.substring`, otherwise we get different behaviour depending on SMP mode.

Reply:
#13784